### ISSUE TITLE
Update UI code of `cli-template-monorepo-ethers` and `cli-template-monorepo-subgraph` 

### ISSUE THREAD

**Issue by vplasencia** [semaphore-protocol/semaphore#836]
## Description

It would be nice to update the `cli-template-monorepo-ethers` and `cli-template-monorepo-subgraph` UI code with the same code as the boilerplate following the React good practices.

Boilerplate UI code: https://github.com/semaphore-protocol/boilerplate/tree/main/apps/web-app

**Comment by yagopajarino** [comment]
Hi @vplasencia, I may be able to contribute with this refactoring. Could you assign it to me?

**Comment by vplasencia** [comment]
Hey @yagopajarino ! Thank you. I just assigned you the issue. Let us know if you have any questions.

**Comment by yagopajarino** [comment]
Hi @vplasencia, I summited the PR but looks like it's failing everywhere. However I tested it locally and worked fine. 

I realized the new UI uses chakra components, therefore some new packages need to be installed but github actions don't allow me to do so.

<img width="1047" alt="image" src="https://github.com/user-attachments/assets/23d41d5d-202a-4a5b-ba02-966074718310">

Let me know what you think and what changes should I do
